From: Jiri Slaby <jslaby@suse.cz>
Subject: mm: convert the rest to new page table lock api (the suse-only cases)
References: fate#315482
Patch-mainline: never, depends on suse patches

This is a continuation of our patch
patches.suse/mm-convert-the-rest-to-new-page-table-lock-api.patch
which is the upstream commit c4088ebdca64c9a2e34a38177d2249805ede1f4b
(mm: convert the rest to new page table lock api).

Since 3.12.39, commit 8a1f8bd0262f701364ae1344cd230ef86d273007
(mm/hugetlb: take page table lock in follow_huge_pmd()), upstream
commit e66f17ff71772b209eed39de35aaa99ba819c93d, we have one more
occurence of &mm->page_table_lock which has to be converted to
pmd_lockptr. Do so.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 mm/hugetlb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/hugetlb.c
+++ b/mm/hugetlb.c
@@ -3564,7 +3564,7 @@ follow_huge_pmd(struct mm_struct *mm, un
 	struct page *page = NULL;
 	spinlock_t *ptl;
 retry:
-	ptl = &mm->page_table_lock;
+	ptl = pmd_lockptr(mm, pmd);
 	spin_lock(ptl);
 	/*
 	 * make sure that the address range covered by this pmd is not
